core/arch/arm/kernel/thread.c

```diff
@@ -609,6 +609,36 @@ size_t thread_stack_size(void)
 	return STACK_THREAD_SIZE;
 }
 
+void get_stack_limits(vaddr_t *start, vaddr_t *end)
+{
+	uint32_t exceptions = thread_mask_exceptions(THREAD_EXCP_FOREIGN_INTR);
+	struct thread_core_local *l = thread_get_core_local();
+	uint32_t cpu_id = get_core_pos();
+	struct thread_ctx *thr = NULL;
+
+	if (l->curr_thread == -1 || (l->flags & THREAD_CLF_TMP)) {
```

Comment:
If I understand it correctly the stack is used as:
If `THREAD_CLF_TMP` is set we're using tmp_stack
If `THREAD_CLF_ABORT` we're using abort_start unless `THREAD_CLF_TMP` is also set.
If no flags at all, thread stack

Reply:
See `el1_sync_abort()`

Reply:
Hmmmm yes, I am willing to trust you on this :wink: 
What about `THREAD_CLF_FIQ` and `THREAD_CLF_IRQ`? If I understand correctly they are always combined with `THREAD_CLF_TMP` so tmp stack is used.

Reply:
Yes, that should be correct.